lib/reek/smells/nested_iterators.rb

```diff
@@ -10,6 +10,16 @@ module Smells
     #
     # See {file:docs/Nested-Iterators.md} for details.
     class NestedIterators < SmellDetector
+      # Struct for conveniently associating iterators with their depth (that is, their nesting).
+      IteratorToDepth = Struct.new :iterator, :depth do
+        include Comparable
+        def <=>(other)
+          depth <=> other.depth
+        end
+      end
```

Comment:
OK, I feel really bad now, but how about renaming the `iterator` field to `exp`? `Iterator` being an `iterator` + `depth` sound trippy.

Also: have you considered moving `NestedIterators#ignored_iterator?` to `Iterator#ignored?` – this means `Iterator` would have to know about `NestedIterators#ignore_iterators`, but this seems doable. I’m OK if the naswer is ‘yes, let’s do it in a separate PR’. :)